de1plus/plugins/DPx_Scheduler/plugin.tcl

```diff
@@ -0,0 +1,356 @@
+                    start_idle
+                }
+            }
+        }
+    }
+
+    proc clear_time { day } {
+        set ::DPx_scheduler_minutes($day) ""
+        set ::DPx_scheduler_times($day) ""
+    }
+    proc checkArr {name} {
+        upvar $name arr
+        if {![info exists arr(key1)]} {
+            return 0
+        } else {
+            return 1
+        }
+    }
+
+    proc set_time { day } {
+        if {$::plugins::DPx_Scheduler::pm == "pm" && $::plugins::DPx_Scheduler::hours != 12} {
+            set sh [expr $::plugins::DPx_Scheduler::hours + 12]
+        } else {
+            set sh $::plugins::DPx_Scheduler::hours
+        }
+        if {$::plugins::DPx_Scheduler::pm == "am" && $sh == 12} {
+            set sh [expr $::plugins::DPx_Scheduler::hours - 12]
+        }
+        set m [expr ($sh * 60) + $::plugins::DPx_Scheduler::minutes]
+        set s [expr $m * 60]
+        set z [clock format [clock seconds] -format {%z}]
+        set sep " "
+        if {[info exists ::DPx_scheduler_minutes($day)] == 1} {
+            if {[lsearch -exact $::DPx_scheduler_minutes($day) $m] >= 0} {
+                set ::plugins::DPx_Scheduler::message "That time already exists"
+                after 1200 {set ::plugins::DPx_Scheduler::message ""}
+            } else {
+                append ::DPx_scheduler_times($day) [clock format $s -format {%I:%M%P} -gmt $z] \r
+                append ::DPx_scheduler_minutes($day) $sep$m$sep
+            }
+        } else {
+            append ::DPx_scheduler_times($day) [clock format $s -format {%I:%M%P} -gmt $z] \r
+            append ::DPx_scheduler_minutes($day) $sep$m$sep
+        }
+    }
+
+    proc load_sched {} {
+        set fn "[plugin_directory]/DPx_Scheduler/schedule.tbd"
+        array set ::DPx_scheduler_minutes [encoding convertfrom utf-8 [read_binary_file $fn]]
+        foreach day {Mon Tue Wed Thu Fri Sat Sun} {
+            if {[info exists ::DPx_scheduler_minutes($day)] == 1} {
+                foreach m $::DPx_scheduler_minutes($day) {
+                    set s [expr $m * 60]
+                    set z [clock format [clock seconds] -format {%z}]
+                    append ::DPx_scheduler_times($day) [clock format $s -format {%I:%M%P} -gmt $z] \r
+                }
+            }
+        }
+    }
+
+    if {[file exist "[plugin_directory]/DPx_Scheduler/schedule.tbd"]} {
+        ::plugins::DPx_Scheduler::load_sched
+    }
+
+    proc save_schedule {} {
+        msg "saving DPx Schedule"
+        set fn "[plugin_directory]/DPx_Scheduler/schedule.tbd"
+        upvar ::DPx_scheduler_minutes item
+        set DPx_shed {}
+        foreach k [lsort -dictionary [array names item]] {
+            set v $item($k)
+            append DPx_shed [subst {[list $k] [list $v]\n}]
+        }
+        write_file $fn $DPx_shed
+    }
+
+    proc minutes_digits {} {
+        if {$::plugins::DPx_Scheduler::minutes >= 50 && $::plugins::DPx_Scheduler::minutes < 60} {
```

Comment:
You can use "variable" here, which shortens these a bit

like this:
https://github.com/decentespresso/de1app/blob/main/de1plus/plugins/visualizer_upload/plugin.tcl#L231

or this
https://github.com/decentespresso/de1app/blob/4b6a2959032aca7962ac2b7565c71389d4a8a4fc/de1plus/history_viewer.tcl#L77